rosbag2_storage_mcap/src/mcap_storage.cpp

```diff
@@ -641,6 +643,27 @@ std::vector<rosbag2_storage::TopicMetadata> MCAPStorage::get_all_topics_and_type
   return out;
 }
 
+void MCAPStorage::get_all_message_definitions(
+  std::vector<rosbag2_storage::MessageDefinition> & definitions)
+{
+  ensure_summary_read();
+  auto schema_map = mcap_reader_->schemas();
+  definitions.clear();
+  definitions.reserve(schema_map.size());
+  for (const auto & [id, schema_ptr] : schema_map) {
+    std::string encoded_message_definition = "";
+    if (schema_ptr->data.size() != 0) {
```

Comment:
Small nitpicks from Clang-Tidy
```suggestion
    std::string encoded_message_definition;
    if (!schema_ptr->data.empty()) {
```